lib/ts/recipe/passwordless/components/features/signInAndUp/index.tsx

```diff
@@ -0,0 +1,257 @@
+                    const loginAttemptInfo = {
+                        ...this.state.loginAttemptInfo,
+                        lastResend: new Date().getTime(),
+                    };
+                    await this.props.recipe.recipeImpl.setLoginAttemptInfo(loginAttemptInfo);
+                    this.setState((os) => ({
+                        ...os,
+                        error: undefined,
+                        loginAttemptInfo,
+                    }));
+                } else if (res.status === "RESTART_FLOW_ERROR") {
+                    await this.props.recipe.recipeImpl.clearLoginAttemptInfo();
+
+                    this.setState((os) => ({
+                        ...os,
+                        error: SIGN_IN_UP_RESEND_RESTART_FLOW_ERROR,
+                        loginAttemptInfo: undefined,
+                    }));
+                } else if (res.status === "GENERAL_ERROR") {
+                    this.setState((os) => ({
+                        ...os,
+                        error: res.message,
+                        loginAttemptInfo: undefined,
+                    }));
+                }
+                return res;
+            },
+
+            consumeCode: async (input) => {
+                const res = await this.props.recipe.recipeImpl.consumeCode(input);
+                if (res.status === "RESTART_FLOW_ERROR") {
+                    await this.props.recipe.recipeImpl.clearLoginAttemptInfo();
+
+                    this.setState((os) => ({
+                        ...os,
+                        error: SIGN_IN_UP_CODE_CONSUME_RESTART_FLOW_ERROR,
+                        loginAttemptInfo: undefined,
+                    }));
+                } else if (res.status === "GENERAL_ERROR") {
+                    this.setState((os) => ({
+                        ...os,
+                        error: res.message,
+                        loginAttemptInfo: undefined,
+                    }));
+                } else if (res.status === "OK") {
+                    await this.props.recipe.recipeImpl.clearLoginAttemptInfo();
+                }
+
+                return res;
+            },
+
+            clearLoginAttemptInfo: async () => {
+                await this.props.recipe.recipeImpl.clearLoginAttemptInfo();
+                this.setState((os) => ({
+                    ...os,
+                    loginAttemptInfo: undefined,
+                    error: undefined,
+                }));
+            },
+        };
+    };
+
+    componentDidMount = async (): Promise<void> => {
+        const loginAttemptInfo = await this.getModifiedRecipeImplementation().getLoginAttemptInfo();
+
+        // We could be using storage events for this, but we need to keep customization in mind.
+        // Someone could be using something else other than localstorage.
+        const checkSessionIntervalHandle = setInterval(async () => {
+            // TODO: check if we can call into the session recipe from here.
```

Comment:
You can remove this TODO. It's safe to assume that we can call the session recipe.